### ISSUE TITLE
Rater does not update when receive new props

### ISSUE THREAD

**Issue by saniko** [NdYAG/react-rater#13]
HI,
When I receive new updated props, rater does not get updated,
all other elements do.
`              <Rater total={5} rating={rating} interactive={false} />
`

Any idea?

**Comment by NdYAG** [comment]
Hi @saniko 
I'll make a demo to check if <Rater /> would update when its props get updated, 
but it's a little weird because it should work...
Could you show me some more specific code?

**Comment by NdYAG** [comment]
Hi @saniko 
I managed to reproduce the bug.
It might be the removal of `componentWillReceiveProps`, I'll make a version later and notify you if it's published. 

Thanks!

**Comment by NdYAG** [comment]
Hi @saniko 
Please update to version 0.4.3. It is fixed.
Thanks again for reporting this bug. :D

**Comment by saniko** [comment]
Thanks, I will check.

**Comment by vukasin-nikodijevic** [comment]
It seems that this issue isn't fully resolved as I have strange behaviour.

Even If I passed properly params somehow provided params and component state doesn't match.

Logic executed in rateing handler if filtered by `type === "click"`.
```
handleRate = ({rating, type}) => {
    if (type === 'click') {
        ........
    }
  };
```
Please note that I didn't click on any of stars.

Issue appears with just mouse over `stars`.

Check attached [screenshot](https://www.awesomescreenshot.com/image/2979359/cb4983020c07c5920619530be194e5a6).

**Comment by NdYAG** [comment]
Hi @vukasin-nikodijevic ,
could you show me some working snippets so that I could reproduce the bug?

**Comment by vukasin-nikodijevic** [comment]
@NdYAG - sure I'll try to create code pen for this issue.

**Comment by vukasin-nikodijevic** [comment]
@NdYAG  here is demo repo with steps to reproduce bug:
https://github.com/vukasin-nikodijevic/react-rater-bug

Demo video:
https://raw.githubusercontent.com/vukasin-nikodijevic/react-rater-bug/master/bug-demo.webm

**Comment by NdYAG** [comment]
@vukasin-nikodijevic I'll check on it and come back soon!

**Comment by NdYAG** [comment]
Hi @vukasin-nikodijevic,
Thank you for reporting this!
It turns out that there are some bugs in `componentWillReceiveProps`,  please take a look of this commit https://github.com/NdYAG/react-rater/commit/0a526ecc78b0adc735992ce989157c936f00194f if you are interested.

You can update to version 0.4.7, but to make it work for your repo, it's necessary to remove the fractional part. As this lib currently doesn't support fractional part in interactive mode. https://github.com/ndyag/react-rater#read-only-mode

```js
      <div className="App">
        <div style={{width: "100%", height: "50px"}}>
          <button onClick={() => {
            this.setState({rating: 1});
          }}>Rate 1
          </button>
          <button onClick={() => {
            this.setState({rating: 2});
          }}>Rate 2
          </button>
          <button onClick={() => {
            this.setState({rating: 3});
          }}>Rate 3
          </button>
          <button onClick={() => {
            this.setState({rating: 4});
          }}>Rate 4
          </button>
          <button onClick={() => {
            this.setState({rating: 5});
          }}>Rate 5
          </button>
        </div>
        <Rater total={5} rating={this.state.rating} onRate={this.ratingHandler.bind(this)}/>
      </div>
```

Basically, it's easy to implement some validation code to check whether the props is a integer or not, but I think some people may tend to treat `2.5` as `2` while others may treat it as `3`. 

So please do not pass a number with fractional part when `interactive` is true.
